test/replication-luatest/gh_7318_box_wait_limbo_acked_test.lua

```diff
@@ -0,0 +1,433 @@
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    local csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_SYNC', true)
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_fiber_csw_geq_than(cg.replica, f, csw + 1)
+    cg.replica:exec(function(f)
+        t.assert_equals(box.info.synchro.queue.len, 1)
+        box.error.injection.set('ERRINJ_WAL_SYNC', false)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        require('fiber').find(f):join()
+    end, {f})
+    server_becomes_the_leader_again(cg.master)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+end
+
+g.test_update_greatest_term_while_wal_sync = function(cg)
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function() box.ctl.promote() end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
+    end)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 1)
+    server_becomes_the_leader_again(cg.master)
+    local csw = get_fiber_csw(cg.replica, f)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_fiber_csw_geq_than(cg.replica, f, csw + 1)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.replica:exec(function(f)
+        box.error.injection.set('ERRINJ_WAL_DELAY', false)
+        require('fiber').find(f):join()
+    end, {f})
+    server_wait_promote_greatest_term_geq_than(cg.replica, 3)
+    server_wait_synchro_queue_len_is_equal(cg.replica, 0)
+end
+
+g.test_txn_limbo_is_empty_while_wal_sync = function(cg)
+    cg.master:exec(function()
+        box.cfg{
+            election_mode='off',
+            replication_synchro_quorum=1
+        }
+    end)
+
+    local f = cg.replica:exec(function()
+        box.error.injection.set('ERRINJ_WAL_DELAY_COUNTDOWN', 0)
+        local f = require('fiber').create(function() box.ctl.promote() end)
+        f:set_joinable(true)
+        return f:id()
+    end)
+    server_wait_wal_is_blocked(cg.replica)
+    cg.master:exec(function()
+        require('fiber').create(function() box.space.test:insert{1} end)
```

Comment:
You're writing on master with quorum 1. The write will succeed right away, no need to create a fiber here.